File: Test_Triangles.py
#!/usr/bin/python3

import unittest
from Triangles import *

class TestTriangles(unittest.TestCase):
    def test_EqIsosceles(self):
        unit = Triangles()
        s = unit.getInfo(1,2,3,4,5,6)
        EXPSTR = "It is an isosceles triangle but not a right triangle"
        self.assertEqual(s, EXPSTR)

    def test_EqScalene(self):
        unit = Triangles()
        s = unit.getInfo(2,2,2,-3,-1,2)
        EXPSTR = "It is a scalene triangle and it is a right triangle"
        self.assertEqual(s, EXPSTR)

if __name__ == '__main__':
    unittest.main()

File: src/Test_Triangles.cpp
#include <gtest/gtest.h>
#include "Triangles.h"
#include <string>

using namespace chain;

TEST(Triangles,TestEqIsosceles) {
    try {
        Triangles(1, 2, 3, 4, 5, 6);
    } catch (std::out_of_range &ex) {
        std::cout << ex.what() << std::endl;
        EXPECT_STREQ("It is an isosceles triangle but not a right triangle.", ex.what());
    }
}

TEST(Triangles,TestEqScalene){
    try {
        Triangles(2,2,2,-3,-1,2);
    } catch (std::out_of_range &ex){
        std::cout<<ex.what()<<std::endl;
        EXPECT_STREQ("It is a scalene triangle and it is a right triangle.",ex.what());
    }
}



